test/cpp/server/credentials_test.cc

```diff
@@ -116,6 +120,102 @@ TEST(
   CHECK_NE(server_credentials.get(), nullptr);
 }
 
+TEST(CredentialsTest,
```

Comment:
I don't think we need to test every success and error case both here and at the C-core layer.  The C-core tests already cover all of the cases; I think the code here should simply check a single success case and a single error case for each of the two provider impls, just to prove that we're propagating the result from the C-core layer correctly.

Reply:
Removed redundant tests.